lib/mergeset/table.go

```diff
@@ -531,7 +531,7 @@ func (tb *Table) AddItems(items [][]byte) {
 //
 // The appended parts must be released with putParts.
 func (tb *Table) getParts(dst []*partWrapper) []*partWrapper {
-	tb.partsLock.Lock()
+	tb.partsLock.RLock()
```

Comment:
Is it actually a hot path for mutex? 

 WDYT about adding readlock for  all read  cases? Especially assistedMergeForInmemoryParts looks like good canditate for read lock.

Reply:
The change is based on the mutex profile I get from one of the cases. The mutex profile shows contention on lock release for code:
1. lib/mergeset.(*Table).getParts
2. lib/mergeset.(*Table).mergeExistingParts

The mutex profile, if we can trust this info, was collected during the incident where VM choked on accepting a lot of new read queries at once. Since `getParts` is invoked during IndexSearch, I assumed it could be a place of contention between background merges and read queries.

Reply:
The `sync.RWMutex` makes sense in cases where the code section under `RLock` takes significant amounts of time (e.g. more than a millisecond). This allows executing the code under the `RLock` section in parallel on multiple CPU cores. In other cases switching from `sync.Mutex` to `sync.RWMutex` doesn't help and even can hurt by increasing code complexity and by reducing its performance on systems with big number of CPU cores, where the time to take `RLock` increases with the number of available CPU cores.

In this case the `RLock` code section is very small, so it must be executed in less than a millisecond in most cases (actually, the code should be executed in less than a microsecond). This means that the switch from `sync.Mutex` to `synx.RWMutex` has little sense here.